### PR TITLE
test: Add coverage for token exchange

### PR DESCRIPTION
**Pull request by ivov**
https://linear.app/n8n/issue/PAY-2247/write-tests-for-launcher

### CODE REVIEW

File: internal/http/fetch_grant_token_test.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+}
+
+func TestFetchGrantTokenInvalidURL(t *testing.T) {
+	_, err := FetchGrantToken("not-a-valid-url", "test-token")
+	if err == nil {
+		t.Error("Expected error for invalid URL, got nil")
+	}
+}
+
+func TestFetchGrantTokenRetry(t *testing.T) {
```
Comment (tomi) [inline internal/http/fetch_grant_token_test.go:126]: Would it be possible to test also failing connection to the server? Feel free to ignore if gets to cumbersome

**Review by tomi — DISMISSED** [review]
🚀

File: internal/http/fetch_grant_token_test.go (hunk 1 above)
Comment (ivov) [reply internal/http/fetch_grant_token_test.go:126]: [84e4798](https://github.com/n8n-io/task-runner-launcher/pull/38/commits/84e4798a0c9e37a9470aeecb3b451ec25f0e7b26)

File: internal/http/check_until_broker_ready_test.go [hunk 2]
```diff
@@ -29,18 +29,18 @@ func TestCheckUntilBrokerReadyHappyPath(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			requestCount := 0
-			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```
Comment (tomi) [inline internal/http/check_until_broker_ready_test.go:32]: Personally I'm not a big fan of abbreviating names too much. There's plenty of room in the world for characters to have clear and explicit names :D

**Review by tomi — APPROVED** [review]
🚀

File: internal/http/check_until_broker_ready_test.go (hunk 2 above)
Comment (ivov) [reply internal/http/check_until_broker_ready_test.go:32]: Agree! Following this as I think it's the Go style.

**Pull request merged**